### PR TITLE
PBL-25234, PBL-25484 can't cope with per-identifier targetPlatforms

### PR DESCRIPTION
**Pull request by Spacerat**
This adds full support (to the beta.cloudpebble.net branch) for resources with multiple identifiers, where each identifier may have a different setting for 'targetPlatforms', bringing CloudPebble closer to fully supporting all facets of Pebble's appinfo.json

The following screenshot shows how the interface is now able to display multiple IDs for a PNG resource, and each ID has a delete button. (fixing "PBL-25484 Cannot delete secondary font configurations in CloudPebble")

screen shot 2015-10-21 at 16 25 08

As with fonts at the moment, a button at the bottom of the page allows users to add a new identifier.

screen shot 2015-10-06 at 08 20 01

(now opened against master!)

### CODE REVIEW

File: ide/static/ide/js/resources.js [hunk 1]
```diff
@@ -357,25 +307,84 @@ CloudPebble.Resources = (function() {
         // Validate the tags!
         var new_tag_values = get_new_tag_values(form, !!file, true);
 
-        // Validate the tags: ensure that all variants are unique
+        if (CloudPebble.ProjectInfo.sdk_version == "2" && new_tag_values.length > 1) {
+            report_error(gettext("SDK 2 projects do not support multiple files per resource. Please delete extra files."))
+            returnl
```
Comment (Katharine) [inline ide/static/ide/js/resources.js]: Typo.

File: ide/static/ide/js/resources.js [hunk 2]
```diff
@@ -357,25 +307,84 @@ CloudPebble.Resources = (function() {
...
+        }
+
+        // Ensure that all variant's tags are unique
```
Comment (Katharine) [inline ide/static/ide/js/resources.js]: `variants'`

File: ide/static/ide/js/resources.js [hunk 3]
```diff
@@ -555,68 +588,76 @@ CloudPebble.Resources = (function() {
...
-                        'letter-spacing': tracking + 'px',
-                        'transform': 'scale(' + (96 / PEBBLE_PPI) + ')',
-                        'transform-origin': '0 0',
-                        'display': 'inline-block',
-                        'border': (2 * (PEBBLE_PPI / 96)) + 'px solid #767676',
-                        'padding': '5px',
-                        'border-radius': '5px',
-                        'background-color': 'white',
-                        'color': 'black'
+                if (resource.kind == 'font') {
```
Comment (Katharine) [inline ide/static/ide/js/resources.js]: Why is this necessary?

Comment (Spacerat) [inline ide/static/ide/js/resources.js]: If I do the check for resource.kind == 'font' before the call, it's not! (Which I've just committed)
